### PR TITLE
attempted fix for key of zero length on Ubuntu 14

### PR DESCRIPTION
**Pull request by edoardocomar**
fixes issues #46 

tested on MacOS as well

developed with @mimaison

### CODE REVIEW

**Comment by edoardocomar** [comment]
@webmakersteve we believe that this approach solves the bizarre issue we encountered on Ubuntu 14 - which is a platform that is essential for us.

File: src/producer.cc [hunk 1]
```diff
@@ -322,18 +322,17 @@ NAN_METHOD(Producer::NodeProduce) {
...
+    key = new std::string(*keyUTF8);
   }
 
   Producer* producer = ObjectWrap::Unwrap<Producer>(info.This());
 
   Baton b = producer->Produce(message_buffer_data, message_buffer_length,
     topic->toRDKafkaTopic(), partition, key);
 
+  // we can delete the key as librdkafka will take a copy of the message
+  if (key) delete key;
```
Comment (webmakersteve) [inline src/producer.cc]: Can you add braces here to make sure it is a multi-line `if`.

e.g.

```
if (key) {
   delete key;
}
```

**Comment by webmakersteve** [comment]
One change and I will merge.

**Comment by webmakersteve** [comment]
Merged! Thanks for the PR.

**Pull request merged**